starknet_py/net/account/base_account.py

```diff
@@ -0,0 +1,173 @@
+from abc import ABC, abstractmethod
+from typing import Optional, List, Union
+
+from starknet_py.net.client import Client
+from starknet_py.net.client_models import (
+    Calls,
+    InvokeFunction,
+    Declare,
+    SentTransactionResponse,
+    EstimatedFee,
+    Hash,
+    Tag,
+)
+from starknet_py.net.models import AddressRepresentation
+from starknet_py.net.models.transaction import DeployAccount
+from starknet_py.utils.typed_data import TypedData
+
+
+class BaseAccount(ABC):
+    @property
+    @abstractmethod
+    def client(self) -> Client:
+        """"""
+        # TODO docstring
+
+    @property
+    @abstractmethod
+    def supported_tx_version(self) -> int:
+        """"""
+        # TODO docstring
+
+    @abstractmethod
+    async def signed_estimate_fee(
```

Comment:
I don't like this method.

Reply:
What alternative do you suggest? We very much need to be able to estimate fee for signed transactions, previously `AccountClient` had an override for `Client.estimate_fee` adding a signature but of course it's not possible now.

Reply:
User can use for instance `sign_invoke_transaction` and provide its output to Client.estimate_fee, right?

Reply:
Yes, they can do that. I think signed estimate_fee was used in some places in `Contract` but that can be replaced with this kind of flow.